### DISCUSSION TITLE
Custom Stream Network for WRF Hydro

### DISCUSSION THREAD

**Discussion by zhong-industries** [NCAR/wrf_hydro_nwm_public#765] [General]
Is it possible to feed in an existing stream network as the routing grid for WRF Hydro? I'm talking about using popular stream networks such as HydroSHEDS or MERIT Hydro. Right now, it seems like WRF HYDRO calculates its own stream networks from the user supplied digital elevation.

**Comment by kmsampson** [comment]
If you are asking about using an existing gridded hydrography dataset as the routing grid in WRF-Hydro, then this cannot be done directly with the existing gridded hydrography data (MERIT or HydroSHEDs). However, I will mention a few ways to try and replicate these datasets in WRF-Hydro, below. 

You are correct that the WRF-Hydro GIS Pre-processing tools calculate a gridded stream network from supplied elevation data. There are several reasons for this. First and foremost, WRF-Hydro supports the 4 coordinate reference systems that are supported by WRF, which are all based on a spheroidal datum. Thus, any supplied elevation dataset or gridded channel network will need to be transformed (reprojected) to those coordinate systems, which is very likely to break any hydrologic enforcement performed in the creation of those hydrography datasets. The WRF-Hydro GIS pre-processing tools were created to resolve this issue, by reprojecting elevation to the WRF grids and then ensuring hydrologic connectivity of the resulting gridded flow network.

There are several ways that you can try and re-create gridded flow networks similar to the global hydrography datasets you referenced. First, you can use the hydro-conditioned DEMs from those datasets, and ensure that your WRF-Hydro grid is as similar as possible to the coordinate system and grid spacing as those DEMs. This will limit the errors introduced in the reprojection step. Second, you can provide network seed points generated from your input hydrography as inputs to the GIS Pre-processing tools, which should closely match the network density in those datasets.

For Muskingum-Cunge reach-based routing, there are methods to bring in an existing vector hydrography dataset for overland routing in WRF-Hydro. We typically reference the NOAA National Water Model configuration, which uses the NHDPlus vector hydrography network. For this, users would need to calculate and supply spatial weights that correspond to the intersection of catchments in the hydrography dataset and the WRF-Hydro grid cells. These weights are stored in a "UDMP" (user-defined mapping) file. Unfortunately, we do not currently have documentation or tools provided to build these mapping files, so I would point you to the NOAA National Water Model configuration files to find the formatting information for those.

Thanks
